internal/archive/archive.go

```diff
@@ -184,6 +230,13 @@ func openUbuntu(options *Options) (Archive, error) {
 					return nil, err
 				}
 				release = index.release
```

Comment:
It’s because the archive can have multiple suites, `jammy`, `noble` etc. An index is the one that pinpoints which suite to look at to fetch the release. For `supportsArch`, you need the InRelease file which requires index because of the suite to grab it from. For `checkComponents`, it is the same – `index` points to a suite and components need a suite to exist within.

Reply:
You are right, thanks! I was incorrectly thinking there was only one suite. I guess the proper thing would be for these methods to be at the suite-level and not at the component level, but we do not even have that abstraction at the moment.